includes/class-wc-stripe-payment-method-configurations.php

```diff
@@ -58,29 +58,47 @@ class WC_Stripe_Payment_Method_Configurations {
 	 * @return object|null
 	 */
 	private static function get_primary_configuration( $force_refresh = false ) {
-		if ( ! $force_refresh ) {
+		// Only allow fetching payment configuration once per minute.
+		$fetch_cooldown = defined( 'IS_TESTING' ) ? 0 : get_option( 'wcstripe_payment_method_config_fetch_cooldown', 0 );
+		$is_in_cooldown = $fetch_cooldown > time();
+		if ( ! $force_refresh || $is_in_cooldown ) {
 			$cached_primary_configuration = self::get_payment_method_configuration_from_cache();
 			if ( $cached_primary_configuration ) {
 				return $cached_primary_configuration;
 			}
+
+			// If we are hitting the API too much, and our main cache is not working, use the fallback cache.
+			if ( $is_in_cooldown ) {
+				$fallback_cache = self::get_payment_method_configuration_from_cache( true );
+				if ( $fallback_cache ) {
+					return $fallback_cache;
+				}
+			}
```

Comment:
I think it's worth adding an explicit comment to explain that we are intentionally falling through here, and what we expect to happen.
```suggestion
			}
			// Intentionally fall through to fetching the data from Stripe if we don't have it locally, 
			// even when $force_refresh = false and/or $is_in_cooldown is true.
			// We _need_ the payment method configuration for things to work as expected, 
			// so we will fetch it if we don't have anything locally.
```